File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/PauseIndexing.py
from ravendb import StopIndexingOperation, GetIndexingStatusOperation, IndexingStatus
from ravendb.documents.indexes.definitions import IndexRunningStatus
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class Foo:
    # region syntax
    # class name has "Stop", but this is ok, this is the "Pause" operation
    class StopIndexingOperation(VoidMaintenanceOperation):
        def __init__(self): ...

    # endregion


class PauseIndexing(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_pause_indexing(self) -> None:
        with self.embedded_server.get_document_store("PauseIndexing") as store:
            # region pause_indexing
            # Define the pause indexing operation
            pause_indexing_op = StopIndexingOperation()

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(pause_indexing_op)

            # At this point:
            # All indexes in the default database will be 'paused' on the preferred node

            # Can verify indexing status on the preferred node by sending GetIndexingStatusOperation
            indexing_status = store.maintenance.send(GetIndexingStatusOperation())
            self.assertEqual(indexing_status.status, IndexRunningStatus.PAUSED)
            # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/PauseIndexing.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use PHPUnit\Framework\TestCase;
use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexingStatus;
use RavenDB\Documents\Operations\Indexes\GetIndexingStatusOperation;
use RavenDB\Documents\Operations\Indexes\StopIndexingOperation;

class PauseIndexing extends TestCase
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region pause_indexing
            // Define the pause indexing operation
            $pauseIndexingOp = new StopIndexingOperation();

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($pauseIndexingOp);

            // At this point:
            // All indexes in the default database will be 'paused' on the preferred node

            // Can verify indexing status on the preferred node by sending GetIndexingStatusOperation
            /** @var IndexingStatus $indexingStatus */
            $indexingStatus = $store->maintenance()->send(new GetIndexingStatusOperation());
            $this->assertTrue($indexingStatus->getStatus()->isPaused());
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    // class name begins with "Stop" but this is still the "Pause" operation
    StopIndexingOperation()
    # endregion
}
*/
